plugins/core/src/main/kotlin/de/fayard/refreshVersions/core/internal/MavenDependencyVersionsFetcherGCS.kt

```diff
@@ -0,0 +1,61 @@
+package de.fayard.refreshVersions.core.internal
+
+import de.fayard.refreshVersions.core.DependencyVersionsFetcher
+import de.fayard.refreshVersions.core.ModuleId
+import de.fayard.refreshVersions.core.Version
+import de.fayard.refreshVersions.core.extensions.okhttp.await
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import retrofit2.HttpException
+import retrofit2.Response
+import java.text.SimpleDateFormat
+import java.util.*
+
+internal class MavenDependencyVersionsFetcherGCS(
+    moduleId: ModuleId,
+    repoUrl: String
+) : DependencyVersionsFetcher(
+    moduleId = moduleId,
+    repoKey = repoUrl
+) {
+
+    val gcs = GCSFetcher(repoUrl)
+
+    init {
+//        require(repoUrl.endsWith('/'))
+    }
+
+    override suspend fun getAvailableVersions(versionFilter: ((Version) -> Boolean)?): SuccessfulResult {
+        val xml: String = gcs.getBlob("${moduleId.group!!.replace('.', '/')}/${moduleId.name}/maven-metadata.xml")
+
+        val allVersions = parseVersionsFromMavenMetaData(xml)
+
+        return SuccessfulResult(
+            lastUpdateTimestampMillis = parseLastUpdatedFromMavenMetaData(xml),
+            availableVersions = if (versionFilter == null) {
+                allVersions
+            } else {
+                allVersions.filter(versionFilter)
+            }
+        )
+    }
+
+    companion object {
```

Comment:
Isn't that code a duplicate of the standard MavenDependencyVersionsFetcher?

Reply:
it is, and thats why i would call it a hack until i refactor it
i image standard MavenDependencyVersionsFetcher can just be implemented in 3 ways
1. httpCLient
2. reading files (for mavenLocal)
3. using gcn

Reply:
now i refactored it so that there should not be duplicate